File: swagger/models/package.py
from .base_model import BaseModel


class Package(BaseModel):
	"""
	no description available

	Attributes:

		id
		The ID of the package.

		name
		The name of the package.

		discount_percentage
		The discount percentage applied to the package.

		sell_online
		When `true`, only products that can be sold online are returned.<br />
		When `false`, all products are returned.

		services
		Information about the services in the packages.

		products
		Information about the products in the packages.
	"""

	input_map = {
		'Id': 'id',
		'Name': 'name',
		'DiscountPercentage': 'discount_percentage',
		'SellOnline': 'sell_online',
		'Services': 'services',
		'Products': 'products',
		}

	output_map = {
		'id': 'Id',
		'name': 'Name',
		'discount_percentage': 'DiscountPercentage',
		'sell_online': 'SellOnline',
		'services': 'Services',
		'products': 'Products',
		}

	id = None
	name = None
	discount_percentage = None
	sell_online = None
	services = None
	products = None

File: swagger/Models/Package.php
<?php

namespace App\MindBody\Models;

/**
 * no description available
 */
class Package extends BaseModel
{
	/**
	 * The ID of the package.
	 * 
	 * @var integer format:int32
	 */
	public $id;
	/**
	 * The name of the package.
	 * 
	 * @var string 
	 */
	public $name;
	/**
	 * The discount percentage applied to the package.
	 * 
	 * @var number format:double
	 */
	public $discountPercentage;
	/**
	 * When `true`, only products that can be sold online are returned.<br />
	 * When `false`, all products are returned.
	 * 
	 * @var boolean 
	 */
	public $sellOnline;
	/**
	 * Information about the services in the packages.
	 * 
	 * @var Service[] 
	 */
	public $services;
	/**
	 * Information about the products in the packages.
	 * 
	 * @var Product[] 
	 */
	public $products;

	protected function getInputMap()
	{
		return [
			'Id' => ['id', 'integer', 'int32'],
			'Name' => ['name', 'string', null],
			'DiscountPercentage' => ['discountPercentage', 'number', 'double'],
			'SellOnline' => ['sellOnline', 'boolean', null],
			'Services' => ['services', 'array', Service::class],
			'Products' => ['products', 'array', Product::class],
			];
	}

	protected function getOutputMap()
	{
		return [
			'id' => ['Id', 'integer', 'int32'],
			'name' => ['Name', 'string', null],
			'discountPercentage' => ['DiscountPercentage', 'number', 'double'],
			'sellOnline' => ['SellOnline', 'boolean', null],
			'services' => ['Services', 'array', Service::class],
			'products' => ['Products', 'array', Product::class],
			];
	}
}
